test/api/core/pluginsManager/init.test.js

```diff
@@ -79,6 +79,78 @@ describe('PluginsManager', () => {
       should(pluginsManager.plugins).be.empty();
     });
 
+    it('should warn if manifest.json does not define customs errors', () => {
+      fsStub.readdirSync.returns(['kuzzle-plugin-test']);
+      fsStub.statSync.returns({
+        isDirectory: () => true
+      });
+      mockrequire('/kuzzle/plugins/enabled/kuzzle-plugin-test', pluginStub);
+      mockrequire(
+        '/kuzzle/plugins/enabled/kuzzle-plugin-test/manifest.json',
+        { name: 'kuzzle-plugin-test', kuzzleVersion: '^1.x' });
+      
+      pluginsManager.init();
+      should(kuzzle.log.warn)
+        .calledOnce()
+        .calledWith('[kuzzle-plugin-test] This plugin doesn\'t have customs errors in the manifest file.');
+    });
+
+    it('should properly load customs errors from manifest.json', () => {
+      fsStub.readdirSync.returns(['kuzzle-plugin-test']);
+      fsStub.statSync.returns({
+        isDirectory: () => true
+      });
+      mockrequire('/kuzzle/plugins/enabled/kuzzle-plugin-test', pluginStub);
+      mockrequire(
+        '/kuzzle/plugins/enabled/kuzzle-plugin-test/manifest.json',
+        {
+          name: 'kuzzle-plugin-test',
+          kuzzleVersion: '^1.x',
+          errors: {
+            'some_error': {
+              'code': 1,
+              'message': 'Some error occured %s',
+              'class': 'BadRequestError'
+            },
+            'some_other_error': {
+              'code': 2,
+              'message': 'Some other error occured %s',
+              'class': 'ForbiddenError'
+            }
+          }
+        });
+      pluginsManager.init();
+      should(kuzzle.log.info)
```

Comment:
Shouldn't you check that the `plugins` errors are successfully expanded with manifest errors ?

Reply:
I can't see how to do that without changing the implementation of the errors manager:
the json object you want me to check is only exposed in it.

Reply:
But here, the test is only successful if errors are correctly loaded since its done in a try catch
`kuzzle.log.info` is only called if nothing is thrown

Reply:
You can do that with `rewire`.
```
const
  rewire = require('rewire'),
  errors = rewire('./lib/config/error-codes/index');

console.log(errors.__get__('plugins'))
```
I can see that if the console.log is not executed it means that the function fail somewhere but it's more robust (:muscle:) to really test the output instead of some side effect

Reply:
It should be robust like : ![](https://github.githubassets.com/images/icons/emoji/unicode/1f4aa.png)
By now